### PR TITLE
Text collection filter

### PR DESCRIPTION
**Pull request by jmleroux**

### CODE REVIEW

File: src/ArrayConverter/StandardToFlat/Product/ValueConverter/TextCollectionConverter.php [hunk 1]
```diff
@@ -7,7 +7,11 @@
 use Pim\Component\Connector\ArrayConverter\StandardToFlat\Product\ValueConverter\ValueConverterInterface;
 
 /**
- * @author JM Leroux <jean-marie.leroux@akeneo.com>
+ * Standard to flat converter
```
Comment (fitn) [inline src/ArrayConverter/StandardToFlat/Product/ValueConverter/TextCollectionConverter.php]: Please explain a bit more or remove

File: src/Filter/ProductValue/TextCollectionFilter.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+namespace Pim\Bundle\ExtendedAttributeTypeBundle\Filter\ProductValue;
+
+use Oro\Bundle\FilterBundle\Form\Type\Filter\FilterType;
+use Oro\Bundle\FilterBundle\Form\Type\Filter\TextFilterType;
+use Pim\Bundle\ExtendedAttributeTypeBundle\Form\Type\Filter\TextCollectionFilterType;
+use Pim\Bundle\FilterBundle\Filter\ProductValue\StringFilter;
+use Pim\Component\Catalog\Query\Filter\Operators;
+
+/**
+ * TextCollectionFilter
```
Comment (fitn) [inline src/Filter/ProductValue/TextCollectionFilter.php]: Explain a bit more where it is used etc.
Imho, this one should be in a `Datagrid/Form/Type/` directory. Here it is a form type you added in ProductValue so it is really confusing :)

Comment (fitn) [inline src/Filter/ProductValue/TextCollectionFilter.php]: It was so confusing that I am wrong ^^ The class just after should be in the directory I said. I don't know this one. I think it is linked to datagrid so at least a datagrid directory

File: src/Form/Type/Filter/TextCollectionFilterType.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+namespace Pim\Bundle\ExtendedAttributeTypeBundle\Form\Type\Filter;
+
+use Oro\Bundle\FilterBundle\Form\Type\Filter\FilterType;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * TextCollectionFilterType to display the grid filter
```
Comment (fitn) [inline src/Form/Type/Filter/TextCollectionFilterType.php]: :+1:

File: src/Provider/FilterProvider.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+namespace Pim\Bundle\ExtendedAttributeTypeBundle\Provider;
+
+use Pim\Bundle\EnrichBundle\Provider\Filter\FilterProviderInterface;
+use Pim\Bundle\ExtendedAttributeTypeBundle\AttributeType\ExtendedAttributeTypes;
+use Pim\Component\Catalog\Model\AttributeInterface;
+
+/**
+ * Filter provider for attributes
+ *
+ * @author    Julien Sanchez <julien@akeneo.com>
```
Comment (fitn) [inline src/Provider/FilterProvider.php]: Not sure :D

File: src/Provider/FilterProvider.php [hunk 5]
```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Pim\Bundle\ExtendedAttributeTypeBundle\Provider;
+
+use Pim\Bundle\EnrichBundle\Provider\Filter\FilterProviderInterface;
+use Pim\Bundle\ExtendedAttributeTypeBundle\AttributeType\ExtendedAttributeTypes;
+use Pim\Component\Catalog\Model\AttributeInterface;
+
+/**
+ * Filter provider for attributes
```
Comment (fitn) [inline src/Provider/FilterProvider.php]: Same about comments

File: src/Resources/config/datagrid/filters.yml [hunk 6]
```diff
@@ -1,17 +1,11 @@
...
-#            - '@form.factory'
-#            - '@pim_filter.product_utility'
-#        tags:
-#            - { name: oro_filter.extension.orm_filter.filter, type: product_value_range }
-#
-#    pim_extended_attribute_type.datagrid.filter.form.type.range:
-#        class: '%pim_extended_attribute_type.datagrid.filter.form.type.range.class%'
-#        tags:
-#            - { name: form.type,  alias: pim_extended_attribute_type_range_filter }
+    pim_filter.text_collection_filter:
```
Comment (fitn) [inline src/Resources/config/datagrid/filters.yml]: `pim_filter`?
Don't know if it is a typo or not. I let you determine if yes or no. Just from my point of view, it is not a class BC Breakable class (it is but it is not ^^)

File: src/Resources/config/form_extensions/filters.yml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+extensions:
+    akeneo-attribute-text-collection-filter:
+        module: pim/filter/attribute/string
+        config:
+            operators:
+                - CONTAINS
+                - DOES NOT CONTAIN
+                - EMPTY
+                - NOT EMPTY
```
Comment (fitn) [inline src/Resources/config/form_extensions/filters.yml]: Is it really implemented? Not empty seems not implemented in the form type.

Comment (jmleroux) [inline src/Resources/config/form_extensions/filters.yml]: It is not :+1: 
and i'd like an error like this break the tests.
Work in progress

**Pull request merged**